src/geo.c

```diff
@@ -274,17 +261,26 @@ int geoGetPointsInRange(robj *zobj, double min, double max, GeoShape *shape, geo
 
         sptr = lpNext(zl, eptr);
         while (eptr) {
+            double xy[2];
+            double distance = 0;
             score = zzlGetScore(sptr);
 
             /* If we fell out of range, break. */
             if (!zslValueLteMax(score, &range))
                 break;
 
             vstr = lpGetValue(eptr, &vlen, &vlong);
-            member = (vstr == NULL) ? sdsfromlonglong(vlong) :
-                                      sdsnewlen(vstr,vlen);
-            if (geoAppendIfWithinShape(ga,shape,score,member)
-                == C_ERR) sdsfree(member);
+            if (geoWithinShape(shape,score, xy, &distance)
+                == C_OK) {
+                /* Append the new element. */
+                geoPoint *gp = geoArrayAppend(ga);
+                gp->longitude = xy[0];
+                gp->latitude = xy[1];
+                gp->dist = distance;
+                gp->member = (vstr == NULL) ? sdsfromlonglong(vlong) :
+                         sdsnewlen(vstr,vlen);
+                gp->score = score;
```

Comment:
maybe we wanna move these into geoArrayAppend?
i.e. it'll not just return the new pointer, but also take the arguments to to put in it.
this way the call will probably be just one line, and we avoid repeating these 5 twice.